tests/test_main.py

```diff
@@ -41,55 +44,118 @@ def test_prefix_is_not_baseprefix(self):
         self.assertTrue(resp)
 
 
+class DepsGatheringTestCase(unittest.TestCase):
+    """Tests for the gathering stage of consolidate_dependencies."""
+
+    def test_needs_ipython(self):
+        d = main.consolidate_dependencies(needs_ipython=True, child_program=None,
+                                          requirement_files=None, manual_dependencies=None)
+
+        self.assertDictEqual(d, {'pypi': {Requirement.parse('ipython')}})
+
+    def test_child_program(self):
+        child_program = generate_test_file(self, ['"""fades:', 'dep', '"""'])
```

Comment:
Probably it's easier/cleaner to just use `tests/test_files/req_module.py` here?

Reply:
Yeap, didn't look into those test files.